contracts/rsv/Reserve.sol

```diff
@@ -0,0 +1,293 @@
+     */
+    function approve(address spender, uint256 value)
+        external
+        notPaused
+        returns (bool)
+    {
+        _approve(msg.sender, spender, value);
+        return true;
+    }
+
+    /// Transfer approved tokens from one address to another.
+    /// @param from address The address to send tokens from.
+    /// @param to address The address to send tokens to.
+    /// @param value uint256 The number of attotokens to send.
+    function transferFrom(address from, address to, uint256 value)
+        external
+        notPaused
+        returns (bool)
+    {
+        _transfer(from, to, value);
+        _approve(from, msg.sender, data.allowed(from, msg.sender).sub(value));
+        return true;
+    }
+
+    /// Increase `spender`'s allowance of the sender's tokens.
+    /// @dev From MonolithDAO Token.sol
+    /// @param spender The address which will spend the funds.
+    /// @param addedValue How many attotokens to increase the allowance by.
+    function increaseAllowance(address spender, uint256 addedValue)
+        external
+        notPaused
+        returns (bool)
+    {
+        _approve(msg.sender, spender, data.allowed(msg.sender, spender).add(addedValue));
+        return true;
+    }
+
+    /// Decrease `spender`'s allowance of the sender's tokens.
+    /// @dev From MonolithDAO Token.sol
+    /// @param spender The address which will spend the funds.
+    /// @param subtractedValue How many attotokens to decrease the allowance by.
+    function decreaseAllowance(address spender, uint256 subtractedValue)
+        external
+        notPaused
+        returns (bool)
+    {
+        _approve(msg.sender, spender, data.allowed(msg.sender, spender).sub(subtractedValue));
+        return true;
+    }
+
+    /// Mint `value` new attotokens to `account`.
+    function mint(address account, uint256 value)
+        external
+        notPaused
+        only(minter)
+    {
+        require(account != address(0), "can't mint to address zero");
+
+        totalSupply = totalSupply.add(value);
+        require(totalSupply < maxSupply, "max supply exceeded");
+        data.addBalance(account, value);
+        emit Transfer(address(0), account, value);
+    }
+
+    /// Burn `value` attotokens from `account`, if sender has that much allowance from `account`.
+    function burnFrom(address account, uint256 value)
+        external
+        notPaused
+        only(minter)
+    {
+        _burn(account, value);
+        _approve(account, msg.sender, data.allowed(account, msg.sender).sub(value));
+    }
+
+    /// @dev Transfer of `value` attotokens from `from` to `to`.
+    /// Internal; doesn't check permissions.
+    function _transfer(address from, address to, uint256 value) internal {
+        require(to != address(0), "can't transfer to address zero");
+        data.subBalance(from, value);
+        uint256 fee = 0;
+
+        // This is a first-pass implementation and could be totally broken
```

Comment:
I'm not sure this is better. If I want to send my entire balance to someone, I don't want to have to calculate what tx fee tier I'm in, calculate the expected nominal amount, and subtract that from the total transfer amount. It's too much work. I expect if we go this route people will not like it.

Reply:
Sure, but how often do you want to send your entire balance, instead of to send a little bit?

... ok, actually kind of often. Hm. Seems like these might be two pretty separate use cases; maybe they need two separate external functions. Will leave it open for now.